src/plugins/memdump/memdump.h

```diff
@@ -105,10 +105,96 @@
 #ifndef MEMDUMP_H
 #define MEMDUMP_H
 
+#include <vector>
+
 #include <glib.h>
 #include "plugins/private.h"
 #include "plugins/plugins_ex.h"
 
+template<typename T>
+struct call_result_t : public plugin_params<T>
+{
+    call_result_t(T* src) : plugin_params<T>(src), target_cr3(), target_thread(), target_rsp() {}
+
+    void set_result_call_params(const drakvuf_trap_info_t* info, addr_t thread)
+    {
+        target_thread = thread;
+        target_cr3 = info->regs->cr3;
+        target_rsp = info->regs->rsp;
+    }
+
+    bool verify_result_call_params(const drakvuf_trap_info_t* info, addr_t thread)
+    {
+        return (info->regs->cr3 != target_cr3 ||
```

Comment:
Stylistically I would prefer if this to be an `if (...) return false` with a default case of `return true`. Just easier to read.

Reply:
I forgot that this piece was burrowed from `procmon` plugin:
https://github.com/tklengyel/drakvuf/blob/master/src/plugins/procmon/procmon.cpp#L120

Perhaps should be reformatted according to your suggestion and moved to `src/plugins/plugins_ex.h`, as we have at least two plugins which take advantage of it?

Reply:
SGTM

Reply:
Moved to `plugins_ex.h`, was also copy-pasted to `wmimon`.